libsolidity/interface/Natspec.h

```diff
@@ -72,6 +72,13 @@ class Natspec
 	/// @return      A JSON representation
 	///              of a method's return notice documentation
 	static Json::Value extractReturnParameterDocs(std::multimap<std::string, DocTag> const& _tags, std::vector<std::string> const& _returnParameterNames);
+
+	/// Temporary function until https://github.com/ethereum/solidity/issues/11114 is implemented.
+	/// @return all events defined in the contract and its base contracts and all events
+	/// that are emitted during the execution of the contract, but allowing only unique signatures.
+	/// In case of conflict between a library event and a contract one, selects the latter
+	/// In case of conflict between two library events, selects the one that is defined first
```

Comment:
This last sentence is actually not how it was supposed to work. It was supposed to always select natspec from the contract in case of a conflict. If we have Natspec only in libraries, we ignore it. The idea was that it's hard to tell which one is more relevant, so it's better to just omit it completely than choose at random.

I thought you actually got that right in tests, but looking again, it's wrong there too. When you have event with same signature in two libraries, it should not appear in natspec.

Reply:
Yes, I was under the impression that we still would show in this case. 
I will change this.

Reply:
There's also one more reason.

Apart from the choice between two libraries being completely arbitrary, the choice could change due to factors that may seem random to the user. That's because he order of events detected by function call graph is based on AST IDs: https://github.com/ethereum/solidity/blob/0cb279494a9af0938dfbf2ca5d0b21115198e8bb/libsolidity/ast/CallGraph.h#L69

So while you'll get the same event from the same code, when you move things around enough that AST IDs shift, your natspec can just suddenly switch to the other one.

Reply:
Fixed.